src/test/java/com/notably/logic/parser/DeleteCommandParserTest.java

```diff
@@ -1,32 +1,88 @@
-// * therefore should be covered by the ParserUtilTest.
-// */
-//public class DeleteCommandParserTest {
-//
-//    private DeleteCommandParser parser = new DeleteCommandParser();
-//
-//    @Test
-//    public void parse_validArgs_returnsDeleteCommand() {
-//        assertParseSuccess(parser, "1", new DeleteCommand(INDEX_FIRST_PERSON));
-//    }
-//
-//    @Test
-//    public void parse_invalidArgs_throwsParseException() {
-//        assertParseFailure(parser, "a", "MESSAGE_INVALID_COMMAND_FORMAT");
-//    }
-//}
+package com.notably.logic.parser;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.notably.commons.path.AbsolutePath;
+import com.notably.logic.commands.DeleteCommand;
+import com.notably.logic.parser.exceptions.ParseException;
+import com.notably.model.Model;
+import com.notably.model.ModelManager;
+import com.notably.model.block.BlockImpl;
+import com.notably.model.block.BlockModel;
+import com.notably.model.block.BlockModelImpl;
+import com.notably.model.block.Title;
+import com.notably.model.suggestion.SuggestionModel;
+import com.notably.model.suggestion.SuggestionModelImpl;
+import com.notably.model.viewstate.ViewStateModel;
+import com.notably.model.viewstate.ViewStateModelImpl;
+
+class DeleteCommandParserTest {
+
+    private static AbsolutePath toBlock;
+    private static AbsolutePath toAnother;
+    private static AbsolutePath toAnotherBlock;
+    private static Model model;
+    private static DeleteCommandParser deleteCommandParser;
+
+    @BeforeEach
+    public void setUp() {
+        // Set up paths
+        toBlock = AbsolutePath.fromString("/block");
+        toAnother = AbsolutePath.fromString("/another");
+        toAnotherBlock = AbsolutePath.fromString("/another/block");
+
+        // Set up model
+        BlockModel blockModel = new BlockModelImpl();
+        SuggestionModel suggestionModel = new SuggestionModelImpl();
+        ViewStateModel viewStateModel = new ViewStateModelImpl();
+        model = new ModelManager(blockModel, suggestionModel, viewStateModel);
+        deleteCommandParser = new DeleteCommandParser(model);
+
+        // Populate model with test data
+        model.addBlockToCurrentPath(new BlockImpl(new Title("CS2103")));
+        model.addBlockToCurrentPath(new BlockImpl(new Title("another")));
+
+        model.setCurrentlyOpenBlock(toAnother);
+        model.addBlockToCurrentPath(new BlockImpl(new Title("block")));
+        model.addBlockToCurrentPath(new BlockImpl(new Title("CS2103")));
+        model.addBlockToCurrentPath(new BlockImpl(new Title("toAnother")));
+    }
+
+    @Test
+    void parse_relativePathWithPrefix_validDeleteCommand() throws ParseException {
+        final DeleteCommand deleteCommand = deleteCommandParser.parse(" -t block").get(0);
+
+        assertNotNull(deleteCommand);
+    }
```

Comment:
For all the test cases, instead of just do a nullity check on the command, how abt testing the command's behaviour?

Reply:
Do you think i should leave it up to the respective CommandTest(for this case `DeleteCommandTest` to test the behaviour and just check for nullity here ? If i check on both sides, would it be redundant?

Reply:
I feel it is not redundant as we are making sure that the command really works as intended. Because the scope of test cases in the `CommandParser` is larger than the `Command` itself (in terms of variations of input), we cannot really test some parts in the `Command` test files. So it's safer to test the command behaviour in the `Parser` test cases too. Lmk what you think!